src/components/ModalWithForm.js

```diff
@@ -26,6 +26,17 @@ export class ModalWithForm extends Modal {
     super.setEventListeners();
     this._form.addEventListener("submit", this._submitEventHandler);
   }
+
+  changeEventListener(callback) {
+    this._submitEventHandler = callback;
+    this.setEventListeners();
```

Comment:
` this.setEventListeners();`

Please, delete it. You add the listeners here (and the parents listeners as well which are not removed). So, you get a **memory leak**


You need to correct your code into this

```js

  setEventListeners() {
    super.setEventListeners();
    this._form.addEventListener("submit", (evt) => {
      this._submitEventHandler(evt)  // here you call the submit handler
    });
  }

  changeEventListener(callback) {
    this._submitEventHandler = callback;  // here you change the submit handler
  }

```